intrahospital_api/services/base/load_utils.py

```diff
@@ -0,0 +1,160 @@
+    )
+
+    if not batch_loads.exists():
+        return True
+
+    current_running = batch_loads.filter(
+        Q(stopped=None) | Q(state=models.BatchPatientLoad.RUNNING)
+    )
+
+    if current_running.count() > 1:
+        # we should never have multiple batches running at the same time
+        raise BatchLoadError(
+            "We appear to have {} concurrent batch loads".format(
+                current_running.count()
+            )
+        )
+
+    last_run_running = current_running.last()
+
+    if last_run_running:
+        time_ago = timezone.now() - last_run_running.started
+
+        if time_ago.seconds < MAX_ALLOWABLE_BATCH_RUN_TIME:
+            logger.info(
+                "batch still running after {} seconds, skipping".format(
+                    time_ago.seconds
+                )
+            )
+            return False
+        else:
+            logger.error(
+                "Previous batch load for {} still running after {} seconds".format(
+                    service_name, MAX_ALLOWABLE_BATCH_RUN_TIME
+                )
+            )
+            return False
+
+    return True
+
+
+def get_batch_start_time(service_name):
+    """
+    If no previous successful batch run has started.
+    Use the started timestamp of the first Initial Patient Load
+
+    If a previous batch load has run and there was an initial patient
+    load that finished within that time and started before it
+    use the start time of the inital patient load.
+
+    In most cases we should be using the start time of the previous
+    BatchPatientLoad
+
+    ie batch A starts at 10:00 and finishes at 10:03
+    batch B will get all data since 10:00 so that nothing is lost
+
+    however...
+    the batches exclude initial patient loads, but usually that's ok
+    but we have extra cracks to paper over...
+
+    a patient load starts during the batch A load, at 9:58. It finishes
+    the db load from the upstream db at 9:59 but is still saving the data
+    to our db at 10:00
+
+    it is excluded from the batch A, so batch B goes and starts its run
+    from 9:59 accordingly.
+
+    """
+    last_successful_run = models.BatchPatientLoad.objects.filter(
+        state=models.BatchPatientLoad.SUCCESS,
+        service_name=service_name
+    ).order_by("started").last()
+
+    if not last_successful_run:
+        return models.InitialPatientLoad.objects.order_by("started").first().started
+
+    long_patient_load = models.InitialPatientLoad.objects.filter(
+        state=models.InitialPatientLoad.SUCCESS
+    ).filter(
+        started__lt=last_successful_run.started
+    ).filter(
+        stopped__gt=last_successful_run.started
+    ).order_by("started").first()
+
+    if long_patient_load:
+        return long_patient_load.started
+    else:
+        return last_successful_run.started
+
+
+def batch_load(service_name):
+    """
+    A decorator that runs the function as batch load
+    ie wrapping it in the batch load decorator
```

Comment:
I'm not really sure what this docstring is trying to say ?

e.g. I know it's a decorator because it has `@wraps` and looks like a decorator?

What does the decorator do?
When do we use it?

It reads like an explanation that assumes I already know what the thing we're explaining is...